data-common.tf

```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2022, 2023 Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+resource "random_id" "state_id" {
+  byte_length = 6
+}
+
+data "oci_identity_availability_domains" "all" {
+  compartment_id = local.compartment_id
+}
+
+locals {
+  # Stable availability domain selection
```

Comment:
Is this selecting the AD or just generating a map for selection? Please clarify the purpose of this. It looks useful and reusable so perhaps other modules can benefit from this.